packages/xstate-store/src/solid.ts

```diff
@@ -0,0 +1,79 @@
+/* @jsxImportSource solid-js */
+import { AnyActorRef } from 'xstate';
+import { createEffect, createSignal, onCleanup } from 'solid-js';
+import { Store, SnapshotFromStore } from './types';
+
+function defaultCompare<T>(a: T | undefined, b: T) {
+  return a === b;
+}
+
+function useSelectorWithCompare<TStore extends Store<any, any>, T>(
+  selector: (snapshot: SnapshotFromStore<TStore>) => T,
+  compare: (a: T | undefined, b: T) => boolean
+): (snapshot: SnapshotFromStore<TStore>) => T {
+  let previous: T | undefined;
+
+  return (state): T => {
+    const next = selector(state);
+
+    if (previous === undefined || !compare(previous, next)) {
+      previous = next;
+    }
+
+    return previous;
+  };
+}
+
+/**
+ * Creates a selector which subscribes to the store and selects a value from the
+ * store's snapshot, using an optional comparison function.
+ *
+ * @example
+ *
+ * ```tsx
+ * import { donutStore } from './donutStore.ts';
+ * import { useSelector } from '@xstate/store/solid';
+ *
+ * function DonutCounter() {
+ *   const donutCount = useSelector(donutStore, (state) => state.context.donuts);
+ *
+ *   return (
+ *     <div>
+ *       <button onClick={() => donutStore.send({ type: 'addDonut' })}>
+ *         Add donut ({donutCount()})
+ *       </button>
+ *     </div>
+ *   );
+ * }
+ * ```
+ *
+ * @param store The store, created from `createStore(…)`
+ * @param selector A function which takes in the snapshot and returns a selected
+ *   value from it
+ * @param compare An optional function which compares the selected value to the
+ *   previously selected value
+ * @returns A read-only signal of the selected value
+ */
+export function useSelector<TStore extends Store<any, any>, T>(
```

Comment:
I initially went with the name `createSelector`, and I might actually prefer it. I'm not sure what the consensus is on naming functions like this in SolidJS. The library itself has `use*` as well as `create*` naming conventions for functions like this.

Reply:
That's fine, whatever is most idiomatic!

Reply:
I'm just noticing the other solid-related code uses the `useActor` or `useMachine` convention so I guess this makes sense.